core/templates/pages/learner-dashboard-page/new-learner-dashboard-page/goals-tab.component.html

```diff
@@ -0,0 +1,401 @@
+<div class="oppia-goals-box">
+  <p class="oppia-goals-title" [innerHTML]="'I18N_LEARNER_DASHBOARD_CURRENT_GOALS_SECTION' | translate">
+  </p>
+  <div *ngIf="currentGoals.length > 0">
+    <div *ngFor="let topic of currentGoals; let i = index">
+      <div class="oppia-current-goals-content">
+        <div class="oppia-arrow-button">
+          <i class="fa fa-caret-right" aria-hidden="true" *ngIf="!currentGoalsStoryIsShown[i]" (click)="toggleStory(i)"></i>
+          <i class="fa fa-caret-down" aria-hidden="true" *ngIf="currentGoalsStoryIsShown[i]" (click)="toggleStory(i)"></i>
+        </div>
+        <div class="oppia-trash-button" (click)="removeFromLearnerGoals(topic.id, topic.name, i)">
+          <i class="fa fa-trash"></i>
+        </div>
+        <p class="oppia-current-goals-topic-name">
+          Learn {{topic.name}}
+        </p>
+        <p class="oppia-goals-story-name">
+          Complete a story in '{{topic.name}}'
+        </p>
+        <div *ngIf="currentGoalsStoryIsShown[i]">
+          <div *ngFor="let story of topic.canonicalStorySummaryDicts" class="oppia-story-summary-card">
+            <oppia-learner-story-summary-tile [storySummary]="story"
+                                              [topicTitle]="topic.name">
+            </oppia-learner-story-summary-tile>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+  <div *ngIf="currentGoals.length == 0" class="oppia-empty-goals-content">
+    <p [innerHTML]="'I18N_LEARNER_DASHBOARD_EMPTY_CURRENT_GOALS_SECTION' | translate"></p>
+  </div>
+</div>
+<div class="oppia-goals-box">
+  <p class="oppia-edit-goals-title" [innerHTML]="'I18N_LEARNER_DASHBOARD_ALL_GOALS_SECTION' | translate"> </p>
+  <span class="oppia-input-box-subtitle">
+    <em>
+      Limit of {{MAX_CURRENT_GOALS_LENGTH}} goals
+    </em>
+  </span>
+  <div class="oppia-edit-goals-subheading">
+    <img [src]="getStaticImageUrl('/learner_dashboard/paw.svg')" alt="paw image" class="oppia-svg-paw-image">
+    <p class="oppia-edit-goals-topic-title">Topic</p>
+    <p class="oppia-edit-goals-learn-title">Learn</p>
+  </div>
+  <div *ngFor="let topic of editGoals; let i = index">
+    <div class="oppia-edit-goals-content">
+      <a [href]="getTopicPageUrl(topic.urlFragment, topic.classroom)" class="oppia-goals-topic-name">
+        {{topic.name}}
+        <p *ngIf="topicBelongToLearntToPartiallyLearntTopics(topic.name)" class="oppia-new-story-content" [innerHTML]="'I18N_LEARNER_DASHBOARD_NEW_STORY_CONTENT' | translate">
```

Comment:
If this doesn't change dynamically during the lifecycle of the app, then precompute this too.

Reply:
Done.